### ISSUE TITLE
git lfs pull stuck

### ISSUE THREAD

**Issue by swolfish** [github/git-lfs#725]
During the `git lfs pull` download gets stuck and after bunch of time I cancelled it (ctrl+c). And then it's not possible to continue/restart the process.
This is the output:
```bash
(10 of 33 files) 4.91 MB / 92.80 MB
Error opening media file buffer.
Error updating the git index

Errors logged to /home/***/.git/lfs/objects/logs/20151009T131458.580210367.log
Use `git lfs logs last` to view the log.
```
Running `git lfs logs last` got this:
```bash
git-lfs/1.0.0 (GitHub; linux amd64; go 1.5.1; git 7379fa9)
git version 2.1.0

$ git-lfs pull
Error opening media file buffer.
Error updating the git index

exit status 128
goroutine 51 [running]:
github.com/github/git-lfs/lfs.Stack(0x0, 0x0, 0x0)
/Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:557 +0x80
github.com/github/git-lfs/commands.logPanicToWriter(0x7f3a1a05d1e8, 0xc8200340b0, 0x7f3a1a05ee80, 0xc8200340a8)
/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:184 +0xf7f
github.com/github/git-lfs/commands.logPanic(0x7f3a1a05ee80, 0xc8200340a8, 0x0, 0x0)
/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:148 +0x421
github.com/github/git-lfs/commands.handlePanic(0x7f3a1a05ee80, 0xc8200340a8, 0x0, 0x0)
/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:123 +0x4e
github.com/github/git-lfs/commands.LoggedError(0x7f3a1a05ee80, 0xc8200340a8, 0x8da8e0, 0x1c, 0x0, 0x0, 0x0)
/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:73 +0x82
github.com/github/git-lfs/commands.Panic(0x7f3a1a05ee80, 0xc8200340a8, 0x8da8e0, 0x1c, 0x0, 0x0, 0x0)
/Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:83 +0x5d
github.com/github/git-lfs/commands.checkoutWithChan(0xc8200fe060)
/Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:228 +0x923
github.com/github/git-lfs/commands.checkoutFromFetchChan.func1(0xc8200fe060, 0xc8200c2290)
/Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:78 +0x21
created by github.com/github/git-lfs/commands.checkoutFromFetchChan
/Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:80 +0x439

ENV:
LocalWorkingDir=/home/***
LocalGitDir=/home/***/.git
LocalGitStorageDir=/home/***/.git
LocalMediaDir=/home/***/.git/lfs/objects
TempDir=/home/***/.git/lfs/tmp
ConcurrentTransfers=3
BatchTransfer=true
```

**Comment by sinbad** [comment]
It was the `git update-index` that failed, but unfortunately we can't tell why. Might be that the index is locked; the fetch and checkout have actually worked, it's just getting git to recognise it in `git status` that didn't.

Please try with the changes in #735 - this makes it use a more useful logged error instead of panicking in that case, and also outputs the result of the `update-index` stdout/stderr to help diagnose it.

**Comment by larsxschneider** [comment]
I ran into a similar error where "git update index" exits with 128:

```
git-lfs/1.1.0 (GitHub; windows amd64; go 1.5.1; git 258acf1)
git version 2.7.0.windows.1

$ git-lfs.exe pull
Error updating the git index


exit status 128
goroutine 225 [running]:
github.com/github/git-lfs/lfs.Stack(0x0, 0x0, 0x0)
        /Users/rick/go/src/github.com/github/git-lfs/lfs/errors.go:557 +0x87
github.com/github/git-lfs/commands.logPanicToWriter(0x3841e8, 0xc0824a6008, 0x34e7b00, 0xc0824a6000)
        /Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:184 +0xf86
github.com/github/git-lfs/commands.logPanic(0x34e7b00, 0xc0824a6000, 0x0, 0x0)
        /Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:148 +0x428
github.com/github/git-lfs/commands.handlePanic(0x34e7b00, 0xc0824a6000, 0x0, 0x0)
        /Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:123 +0x55
github.com/github/git-lfs/commands.LoggedError(0x34e7b00, 0xc0824a6000, 0x967c40, 0x1f, 0xc082c25e70, 0x1, 0x1)
        /Users/rick/go/src/github.com/github/git-lfs/commands/commands.go:73 +0x89
github.com/github/git-lfs/commands.checkoutWithChan(0xc0823f0d80)
        /Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:229 +0xa63
github.com/github/git-lfs/commands.checkoutFromFetchChan.func1(0xc0823f0d80, 0xc082103ad0)
        /Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:78 +0x28
created by github.com/github/git-lfs/commands.checkoutFromFetchChan
        /Users/rick/go/src/github.com/github/git-lfs/commands/command_checkout.go:80 +0x440

ENV:
LocalWorkingDir=C:\Users\schneil\Desktop\cmdline\fusion
LocalGitDir=C:\Users\schneil\Desktop\cmdline\fusion\.git
LocalGitStorageDir=C:\Users\schneil\Desktop\cmdline\fusion\.git
LocalMediaDir=C:\Users\schneil\Desktop\cmdline\fusion\.git\lfs\objects
TempDir=C:\Users\schneil\Desktop\cmdline\fusion\.git\lfs\tmp
ConcurrentTransfers=100
BatchTransfer=true
GIT_TRACE=1
```
(as you can see I experimented with crazy high concurrent transfers... but that shouldn't cause this problem, right?)

**Comment by technoweenie** [comment]
Git LFS should be printing better messages from the `git update-index` error. Fixed in #1400 and shipped in v1.3.0.
